### PR TITLE
camelCase versus under_scores

### PR DESCRIPTION
**Pull request by technosophos**
- [X] I have followed the [code of conduct](https://github.com/hwayne/awesome-cold-showers/blob/master/CODE_OF_CONDUCT.md).
- [X] I have followed the [contributing](https://github.com/hwayne/awesome-cold-showers/blob/master/CONTRIBUTING.md) guidelines and used the proper item format.
- [X] This is not a duplicate.
- [X] The resource is free.
- [X] The resource is rigorous because: (please list) Two academic papers are cited, both published in respected journals.
- [X] The resource is not obnoxious, condescending, etc.
- [X] My writup is respectful of both sides.
- [X] I've added reasonable caveats.
- [X] I've noted if the resource is a video, pdf, etc.

This is what makes your repo fun: I was going to post the Binkley article, and then discovered the newer Sharif article and learned something new!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,6 +72,25 @@ It's great when people get excited about things, but sometimes they get a little
 
 * **Notes:** Includes ethnography on how programmers debug abbreviated code. Link is to the author preprint. 
 
+#### [An Eye Tracking Study on camelCase and under_score Identifier Styles](http://www.cs.kent.edu/~jmaletic/papers/ICPC2010-CamelCaseUnderScoreClouds.pdf)
```
Comment (hwayne) [inline README.md:75]: Could we you mark this as a (PDF)?

**Comment by hwayne** [comment]
Oh my god this looks amazing.

The writeup looks good. I want to read through the paper before merging, though.

**Comment by hwayne** [comment]
Okay read it, two important caveats that should be added:

1. The sample size was only 15 people.
1. "Subjects  were  historically  trained  mostly  in  the  underscore identifier  style  and  were  all  programmers. " 

I also think the information that this is a replication should be moved from "Caveats" to "Notes". It's not information that changes the strength of the paper's findings.

**Comment by jab** [comment]
Kind of surprising to see the style referred to as "under_scores" rather than [snake_case](https://en.wikipedia.org/wiki/Snake_case). Worth renaming this "camelCase vs. snake_case" (despite the paper's title)?

**Comment by politician** [comment]
Those caveats though!

Does it still make sense to be included here?

**Comment by fabianschwarzfritz** [comment]
I have been following this for a while. I doubt a "study" with a sample size of 15 people should be used to derive conclusions about what's better.

The sample size is way to small, this should not be merged.

**Comment by hwayne** [comment]
Guess who didn't get any updates that you made changes!

Me!!!

**Comment by hwayne** [comment]
> I have been following this for a while. I doubt a "study" with a sample size of 15 people should be used to derive conclusions about what's better.
> 
> The sample size is way to small, this should not be merged.

I agree that it's too small to derive any conclusions from, but it's still more work than was put into "Microservices" and "Agile Methods", both of which I added. Also, I really like the work @technosophos put into it. I'll merge this for now but make a note that we might have to schism much later into "rigorous" and "non-rigorous" stuff.

**Pull request merged**